irr/src/CIrrDeviceSDL.cpp

```diff
@@ -267,6 +267,32 @@ int CIrrDeviceSDL::findCharToPassToIrrlicht(uint32_t sdlKey, EKEY_CODE irrlichtK
 	}
 }
 
+u32 CIrrDeviceSDL::getScancodeFromKey(const KeyCode &key) const
+{
+	u32 keynum = 0;
+	if (key.index() == 0) {
+		auto keycode = std::get<EKEY_CODE>(key);
+		for (const auto &entry: KeyMap) {
```

Comment:
(In principle this is screaming for a bidirectional mapping but I don't think this is a performance concern?)

Reply:
IMO this is hardly a performance concern. With the SDL device this is run when the `KeyPress` object is created from the keycode syntax, which mostly happens when the keybindings (and only ones that are configured as keycodes) are read from the settings (i.e. on startup).